salt/utils/network.py

```diff
@@ -1958,3 +1959,33 @@ def parse_host_port(host_port):
             raise ValueError('bad hostname: "{}"'.format(host))
 
     return host, port
+
+
+@jinja_filter("filter_by_networks")
+def filter_by_networks(values, networks=None):
```

Comment:
Very cool changes. Still do not make this parameter optional. The argument about the pillar was discussed in another comment. IMHO this parameter is no optional.

Reply:
Yes, I just read your comment about that after I did the changes. I'll have it required.